packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -547,7 +547,10 @@ export function scheduleUpdateOnFiber(
     } else {
       ensureRootIsScheduled(root, eventTime);
       schedulePendingInteractions(root, lane);
-      if (executionContext === NoContext) {
+      if (
+        (fiber.mode & ConcurrentMode) === NoMode &&
+        executionContext === NoContext
```

Comment:
Nit: Check `executionContext === NoContext` first because in the common case it will be false and the binary expression will short circuit.